src/k8s-engine/hyperkit.ts

```diff
@@ -215,14 +215,11 @@ export default class HyperkitBackend extends events.EventEmitter implements K8s.
   protected get ipAddress(): Promise<string> {
     return (async() => {
       const { driver, defaultArgs } = this.hyperkitArgs;
-      const args = defaultArgs.concat([
-        'ssh', '--', 'ip', '-4', '-o', 'addr', 'show', 'dev', 'eth0'
-      ]);
+      const args = defaultArgs.concat(['ip']);
       const result = await util.promisify(childProcess.execFile)(driver, args);
-      const match = /\binet\s+([0-9.]+)\//.exec(result.stdout);
 
-      if (match) {
-        return match[1];
+      if (/^[0-9.]+$/.test(result.stdout.trim())) {
```

Comment:
You _could_ simplify like this (not sure if the syntax is correct):

```js
    const ip = (await util.promisify(childProcess.execFile)(driver, args)).stdout.trim()
    if (/^[0-9.]+$/.test(ip)) {
        return ip
    }
```

But I don't know if this would be idiomatic Javascript.

Reply:
No, because I use `result.stderr` if that fails, so I need to keep `result` around. (Otherwise I'd have destructured it.)